redash/query_runner/clickhouse.py

```diff
@@ -67,21 +67,25 @@ def _get_tables(self, schema):
         return schema.values()
 
     def _send_query(self, data, stream=False):
-        r = requests.post(
-            self.configuration.get('url', "http://127.0.0.1:8123"),
-            data=data.encode("utf-8"),
-            stream=stream,
-            timeout=self.configuration.get('timeout', 30),
-            params={
-                'user': self.configuration.get('user', "default"),
-                'password':  self.configuration.get('password', ""),
-                'database': self.configuration['dbname']
-            }
-        )
-        if r.status_code != 200:
-            raise Exception(r.text)
-        # logging.warning(r.json())
-        return r.json()
+        url = self.configuration.get('url', "http://127.0.0.1:8123")
+        try:
+            r = requests.post(
+                url,
+                data=data.encode("utf-8"),
+                stream=stream,
+                timeout=self.configuration.get('timeout', 30),
+                params={
+                    'user': self.configuration.get('user', "default"),
+                    'password':  self.configuration.get('password', ""),
+                    'database': self.configuration['dbname']
+                }
+            )
+            if r.status_code != 200:
+                raise Exception(r.text)
+            # logging.warning(r.json())
+            return r.json()
+        except requests.RequestException:
+            raise Exception('Connection error to %s' % url)
```

Comment:
The problem now is that the error message is very light on details. How about something like the following (assuming `e` is the exception):

```
if e.response:
    details = "({}, Status Code: {})".format(e.__class__.__name__, e.response.status_code)
else:
    details = "({})".format(e.__class__.__name__)

message = "Connection error to: {} {}.".format(url, details)
```

Reply:
It sounds good, fixed, thank you. Now exceptions look like:
`Exception: Connection error to: http://127.0.0.1:8123 (ReadTimeout).`
or
`Exception: Connection error to: http://127.0.0.1:8123 (ConnectionError).`